crates/svm-codec/src/api/json/receipt.rs

```diff
@@ -0,0 +1,353 @@
+    match receipt {
+        ReceiptOwned::DeployTemplate(..) => "deploy-template",
+        ReceiptOwned::SpawnApp(..) => "spawn-app",
+        ReceiptOwned::ExecApp(..) => "exec-app",
+    }
+}
+
+fn decode_error(ty: &'static str, err: &ReceiptError, logs: &[Log]) -> Value {
+    let mut json = {
+        match err {
+            ReceiptError::OOG => json!({
+                "err_type": "oog",
+            }),
+            ReceiptError::TemplateNotFound(template_addr) => json!({
+                "err_type": "template-not-found",
+                "template_addr": json::addr_to_str(template_addr.inner()),
+            }),
+            ReceiptError::AppNotFound(app_addr) => json!({
+                "err_type": "app-not-found",
+                "app_addr": json::addr_to_str(app_addr.inner()),
+            }),
+            ReceiptError::CompilationFailed {
+                app_addr,
+                template_addr,
+                msg,
+            } => json!({
+                "err_type": "compilation-failed",
+                "template_addr": json::addr_to_str(template_addr.inner()),
+                "app_addr": json::addr_to_str(app_addr.inner()),
+                "message": msg,
+            }),
+            ReceiptError::InstantiationFailed {
+                app_addr,
+                template_addr,
+                msg,
+            } => json!({
+                "err_type": "instantiation-failed",
+                "template_addr": json::addr_to_str(template_addr.inner()),
+                "app_addr": json::addr_to_str(app_addr.inner()),
+                "message": msg,
+            }),
+            ReceiptError::FuncNotFound {
+                app_addr,
+                template_addr,
+                func_idx,
+            } => json!({
+                "err_type": "function-not-found",
+                "template_addr": json::addr_to_str(template_addr.inner()),
+                "app_addr": json::addr_to_str(app_addr.inner()),
+                "func_index": func_idx,
+            }),
+            ReceiptError::FuncFailed {
+                app_addr,
+                template_addr,
+                func_idx,
+                msg,
+            } => json!({
+                "err_type": "function-failed",
+                "template_addr": json::addr_to_str(template_addr.inner()),
+                "app_addr": json::addr_to_str(app_addr.inner()),
+                "func_index": func_idx,
+                "message": msg,
+            }),
+        }
+    };
+
+    let logs = json::logs_to_json(logs);
+
+    let map: &mut serde_json::Map<String, Value> = json.as_object_mut().unwrap();
+    let mut map: serde_json::Map<String, Value> = std::mem::take(map);
+
+    map.insert("type".into(), Value::String(ty.into()));
+    map.insert("success".into(), Value::Bool(false));
+    map.insert("logs".into(), Value::Array(logs));
+
+    map.into()
+}
+
+fn decode_deploy_template(receipt: &TemplateReceipt, ty: &'static str) -> Value {
```

Comment:
If you're receiving the native data type and returning a JSON, then you are encoding, not decoding, no?

Reply:
I'm expecting an input JSON looking like this:

```json
{
  "data":  'SEE EXPLANATION BELOW' 
}
```

The `data` entry expects in the value **encoded Receipt given as a string**  

For example, The 0xA12D (2 bytes) should be provided as 'A12D' (4 String chars).
So if the original encoded Receipt consumes `N` bytes, the `data` value should be exactly `2 * N` chars.